src/base/PhpBlock.php

```diff
@@ -67,4 +67,32 @@ public function renderAdmin()
         $this->injectorSetup();
         return $this->admin();
     }
+    
+    public function onRegister()
+    {
+        if ($this->getIsCacheEnabled()) {
+            $phpBlockView = $this->getView();
+    
+            $blockId = $this->getEnvOption('id');
+    
+            $cacheKeyAssets = ['blockassets', $blockId];
+            $cacheKeyAssetBundles = ['blockassetbundles', $blockId];
+            
+            $assets = Yii::$app->cache->getOrSet($cacheKeyAssets, [$phpBlockView, 'getBlockAssets'], $this->getCacheExpirationTime());
+            $assetBundles = Yii::$app->cache->getOrSet($cacheKeyAssetBundles, [$phpBlockView, 'getAssetBundleNames'], $this->getCacheExpirationTime());
+
+            PhpBlockView::registerToAppView($assets, $assetBundles);
+        }
+    }
+    
+    public static function onRegisterFromCache($blockId)
```

Comment:
As mentioned below, add onRegisterFromCache() to the interface but not as static method.  Maybe adding some php docs